src/Features/ShippingLabelBanner.php

```diff
@@ -0,0 +1,188 @@
+				class_exists( '\WC_ShipStation_Integration' );
+
+			$this->shipping_label_banner_display_rules =
+				new ShippingLabelBannerDisplayRules(
+					$jetpack_version,
+					$jetpack_connected,
+					$wcs_version,
+					$wcs_tos_accepted,
+					$incompatible_plugins
+				);
+		}
+
+		return $this->shipping_label_banner_display_rules->should_display_banner();
+	}
+
+	/**
+	 * Add metabox to order page.
+	 *
+	 * @param string   $post_type current post type.
+	 * @param \WP_Post $post Current post object.
+	 */
+	public function add_meta_boxes( $post_type, $post ) {
+		$order = wc_get_order( $post );
+		if ( $this->should_show_meta_box() ) {
+			add_meta_box(
+				'woocommerce-admin-print-label',
+				__( 'Shipping Label', 'woocommerce-admin' ),
+				array( $this, 'meta_box' ),
+				null,
+				'normal',
+				'high',
+				array(
+					'context'               => 'shipping_label',
+					'order_id'              => $post->ID,
+					'shippable_items_count' => $this->count_shippable_items( $order ),
+				)
+			);
+			add_action( 'admin_enqueue_scripts', array( $this, 'add_print_shipping_label_script' ) );
+		}
+	}
+
+	/**
+	 * Count shippable items
+	 *
+	 * @param \WC_Order $order Current order.
+	 * @return int
+	 */
+	private function count_shippable_items( \WC_Order $order ) {
+		$count = 0;
+		foreach ( $order->get_items() as $item ) {
+			if ( $item instanceof \WC_Order_Item_Product ) {
+				$product = $item->get_product();
+				if ( $product && $product->needs_shipping() ) {
+					$count += $item->get_quantity();
+				}
+			}
+		}
+		return $count;
+	}
+	/**
+	 * Adds JS to order page to render shipping banner.
+	 *
+	 * @param string $hook current page hook.
+	 */
+	public function add_print_shipping_label_script( $hook ) {
+		$rtl = is_rtl() ? '-rtl' : '';
+		wp_enqueue_style(
+			'print-shipping-label-banner-style',
+			Loader::get_url( "print-shipping-label-banner/style{$rtl}.css" ),
+			array( 'wp-components' ),
+			Loader::get_file_version( 'print-shipping-label-banner/style.css' )
+		);
+
+		wp_enqueue_script(
+			'print-shipping-label-banner',
+			Loader::get_url( 'wp-admin-scripts/print-shipping-label-banner.js' ),
+			array( 'wp-i18n', 'wp-data', 'wp-element', 'moment', 'wp-api-fetch', WC_ADMIN_APP ),
+			Loader::get_file_version( 'wp-admin-scripts/print-shipping-label-banner.js' ),
+			true
+		);
+
+		$payload = array(
+			'nonce'                 => wp_create_nonce( 'wp_rest' ),
+			'baseURL'               => get_rest_url(),
+			'wcs_server_connection' => true,
+		);
+
+		wp_localize_script( 'print-shipping-label-banner', 'wcConnectData', $payload );
+	}
+
+	/**
+	 * Render placeholder metabox.
+	 *
+	 * @param \WP_Post $post current post.
+	 * @param array    $args empty args.
+	 */
+	public function meta_box( $post, $args ) {
+
+		?>
+		<div id="wc-admin-shipping-banner-root" class=" woocommerce <?php echo esc_attr( 'wc-admin-shipping-banner' ); ?>" data-args="<?php echo esc_attr( wp_json_encode( $args['args'] ) ); ?>">
```

Comment:
minor nit but does the extra space before woocommerce in the class list need to be there?

Reply:
no